File: app/src/main/java/uk/co/victoriajanedavis/chatapp/domain/interactors/AcceptReceivedFriendRequest.java
package uk.co.victoriajanedavis.chatapp.domain.interactors;

import javax.inject.Inject;

import io.reactivex.Single;
import io.reactivex.annotations.NonNull;
import uk.co.victoriajanedavis.chatapp.data.repositories.ReceivedFriendRequestRepository;
import uk.co.victoriajanedavis.chatapp.domain.entities.FriendshipEntity;
import uk.co.victoriajanedavis.chatapp.domain.interactors.ReactiveInteractor.SendInteractor;

public class AcceptReceivedFriendRequest implements SendInteractor<String, FriendshipEntity> {

    @NonNull
    private final ReceivedFriendRequestRepository repository;


    @Inject
    public AcceptReceivedFriendRequest(@NonNull final ReceivedFriendRequestRepository repository) {
        this.repository = repository;
    }

    @NonNull
    @Override
    public Single<FriendshipEntity> getSingle(@NonNull String username) {
        return repository.acceptReceivedFriendRequest(username);
    }
}

File: app/src/main/java/uk/co/victoriajanedavis/chatapp/domain/interactors/AcceptReceivedFriendRequest.kt
package uk.co.victoriajanedavis.chatapp.domain.interactors

import io.reactivex.Completable
import javax.inject.Inject

import uk.co.victoriajanedavis.chatapp.data.repositories.ReceivedFriendRequestRepository
import uk.co.victoriajanedavis.chatapp.domain.interactors.ReactiveInteractor.ActionInteractor
import java.util.UUID

class AcceptReceivedFriendRequest @Inject constructor(
    private val repository: ReceivedFriendRequestRepository
) : ActionInteractor<UUID> {

    override fun getActionCompletable(userUuid: UUID): Completable {
        return repository.acceptReceivedFriendRequest(userUuid)
    }
}
